ui/src/components/exerciseCard/exerciseCard.tsx

```diff
@@ -1,22 +1,75 @@
 import React from "react";
-import { Button, Card } from "react-bootstrap";
+import { Button, Card, ToggleButton} from "react-bootstrap";
 import InfoPanel from "../modalInfoPanel/infoPanel";
-import { ExerciseProps } from "../types";
+import { ExerciseProps ,FavoriteProps} from "../types";
+import { FavoriteService } from "../../services/favoriteService";
 
 interface ExerciseCardProps {
   onClick: (e: ExerciseProps) => void;
   exercise: ExerciseProps;
+  checkIfFavorite: boolean;
 }
 
-export function ExerciseCard({ onClick, exercise }: ExerciseCardProps) {
+async function sendFavorite(data: FavoriteProps) {
+  const favoriteService = new FavoriteService();
+  return await favoriteService.createFavorite(data);
+}
+
+async function deleteFavorite(data: FavoriteProps) {
+  const favoriteService = new FavoriteService();
+  return await favoriteService.deleteFavorite(data);
+}
+
+export function ExerciseCard({ onClick, exercise, checkIfFavorite }: ExerciseCardProps) {
   const [modalShow, setModalShow] = React.useState(false);
+  const [checked, setChecked] = React.useState(false);
+  const [favorite, setFavorite] = React.useState<FavoriteProps>({
+    userId: '123',
+    exersiceId: exercise.id,
+    exersiceProps: exercise
+  });
 
+  
   const handleClick = () => {
     onClick(exercise);
   };
 
+  const handleFavorite = () => {
+
+    if(!checked){
+    sendFavorite(favorite).then((data) => {
+      if (data.data.success === true) {
+        console.log("all good");
+      } else if (
+        data.data.success === false &&
+        data.data.status === "Session already exists"
```

Comment:
The backend doesn't return this, it returns Favorite already exists or somethign